bindings/aws/s3/s3.go

```diff
@@ -44,37 +54,83 @@ func (s *AWSS3) Init(metadata bindings.Metadata) error {
 	if err != nil {
 		return err
 	}
-	uploader, err := s.getClient(m)
+	uploader, downloader, svc, err := s.getClient(m)
 	if err != nil {
 		return err
 	}
 	s.metadata = m
 	s.uploader = uploader
+	s.downloader = downloader
+	s.svc = svc
 
 	return nil
 }
 
 func (s *AWSS3) Operations() []bindings.OperationKind {
-	return []bindings.OperationKind{bindings.CreateOperation}
+	return []bindings.OperationKind{bindings.CreateOperation, bindings.GetOperation, bindings.DeleteOperation, CopyOperation}
 }
 
 func (s *AWSS3) Invoke(req *bindings.InvokeRequest) (*bindings.InvokeResponse, error) {
-	key := ""
-	if val, ok := req.Metadata["key"]; ok && val != "" {
-		key = val
-	} else {
-		key = uuid.New().String()
-		s.logger.Debugf("key not found. generating key %s", key)
-	}
+	key := s.getKey(req)
+	
+	switch op := req.Operation; op {
+	case bindings.CreateOperation:
+		r := bytes.NewReader(req.Data)
+		_, err := s.uploader.Upload(&s3manager.UploadInput{
+			Bucket: aws.String(s.metadata.Bucket),
+			Key:    aws.String(key),
+			Body:   r,
+		})
+		if (err != nil) {
+			s.logger.Errorf("s3 CREATE operation exceptions: %s", err)
+		}
+		return nil, err
```

Comment:
A slight behavior enhancement here? Respond with a value instead of nil?
InvokeResponse struct : with Data field either being `req.Data` or empty.
Metadata field with the following values: `{"bucket": bucket, "key": key}` 
@pkedy WDYT?

Reply:
+

Reply:
@pkedy let me know if this change is acceptable change.

Reply:
That's ok, `daprd` is prepared to deal with this.